flex-checks/src/main/java/org/sonar/flex/checks/ConditionalStructure.java

```diff
@@ -0,0 +1,174 @@
+ * SonarQube Flex Plugin
+ * Copyright (C) 2010-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.flex.checks;
+
+import com.sonar.sslr.api.AstNode;
+import com.sonar.sslr.api.AstNodeType;
+import com.sonar.sslr.api.Token;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.sonar.flex.FlexGrammar;
+import org.sonar.flex.FlexKeyword;
+
+import static java.util.Collections.singletonList;
+
+class ConditionalStructure {
+  static class BranchAndContent {
+    AstNode branch;
+    List<AstNode> content;
+    boolean onliner;
+    BranchAndContent(AstNode branch, List<AstNode> content, boolean oneliner) {
+      this.branch = branch;
+      this.content = content;
+      this.onliner = oneliner;
+    }
+  }
+  final List<BranchAndContent> branches;
+  final boolean allBranchesArePresent;
+
+  ConditionalStructure(List<BranchAndContent> branches, boolean allBranchesArePresent) {
+    this.branches = branches;
+    this.allBranchesArePresent = allBranchesArePresent;
+  }
+
+  boolean areAllEquivalentBranches() {
+    BranchAndContent first = branches.get(0);
+    return branches.stream().skip(1).allMatch(next -> SyntacticEquivalence.areEquivalent(first.content, next.content));
+  }
+
+  @FunctionalInterface
+  interface DuplicatedBranchCallback{
+    void onDuplicatedBranch(AstNode branchFirstNode1, AstNode branchFirstNode2);
+  }
+
+  void forEachBranchDuplication(DuplicatedBranchCallback callback) {
+    for (BranchAndContent branch1 : branches) {
+      if (!allBranchesArePresent || !branch1.onliner) {
+        for (BranchAndContent branch2 : branches) {
+          if (branch1 == branch2) {
+            break;
+          }
+          if (SyntacticEquivalence.areEquivalent(branch1.content, branch2.content)) {
+            callback.onDuplicatedBranch(branch1.branch, branch2.branch);
+          }
+        }
+      }
+    }
+  }
+
+  static boolean isOnelinerIf(AstNode node) {
+    AstNodeType type = node.getType();
+    if (type == FlexGrammar.SUB_STATEMENT) {
+      AstNode child, grandchild;
+      if ((child = node.getFirstChild(FlexGrammar.STATEMENT)) != null) {
+        if ((grandchild = child.getFirstChild(FlexGrammar.BLOCK)) != null) {
```

Comment:
I think that you could replace the whole content of the block with:
```
return isOnelinerSwitch(grandchild.getChildren(FlexGrammar.DIRECTIVES));
```

Reply:
It really helped me to understand a few things.
Thanks.